### PR TITLE
Add decks vinyl service

### PR DESCRIPTION
**Pull request by AlexKievDeveloper**

### CODE REVIEW

File: src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+        }
+        log.info("Got row offers from Decks.de: {}", rawOfferSet.size());
+        return rawOfferSet;
+    }
+
+    boolean isValid(RawOffer rawOffer) {
+        boolean isValid = false;
+        if (rawOffer.getPrice() != 0.
+                && rawOffer.getCurrency().isPresent()
+                && !("".equals(rawOffer.getRelease()))
```
Comment (TEdwardJ) [inline src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java]: isEmpty() ?

File: src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                log.warn("Can't fill raw offer by offer link, not adding it to set {'rawOffer':{}, 'offerLink':{}}",
+                        rawOffer, offerLink);
+            }
+        }
+        log.info("Got row offers from Decks.de: {}", rawOfferSet.size());
+        return rawOfferSet;
+    }
+
+    boolean isValid(RawOffer rawOffer) {
+        boolean isValid = false;
```
Comment (TEdwardJ) [inline src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java]: what about direct setting of the variable?
        boolean isValid = rawOffer.getPrice() != 0d
                && rawOffer.getCurrency().isPresent()
                && !rawOffer.getRelease().isEmpty()
                && rawOffer.getOfferLink() != null;

**Review by TEdwardJ — DISMISSED** [review]

File: src/main/resources/templates/profile.html [hunk 3]
```diff
@@ -44,7 +44,7 @@
               <a href="#" class="header__menu-item">stores</a>
```
Comment (Greengrer) [inline src/main/resources/templates/profile.html]: Do you need to change href to stores?

File: src/test/java/com/vinylteam/vinyl/dao/jdbc/mapper/UserPostRowMapperTest.java [hunk 4]
```diff
@@ -7,31 +7,37 @@
...
 
 class UserPostRowMapperTest {
+
     private final RowMapper<UserPost> rowMapper = new UserPostRowMapper();
 
     @Test
     @DisplayName("Map fields correctly for user post conversion")
     void mapFilledRowTest() throws SQLException {
+        LocalDateTime createdAt = LocalDateTime.now();
         //prepare
```
Comment (Greengrer) [inline src/test/java/com/vinylteam/vinyl/dao/jdbc/mapper/UserPostRowMapperTest.java:24]: Should be on line 24.

File: src/test/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    @Test
+    @DisplayName("Checks that service generate not empty array for empty session id")
+    void emptyArrayForEmptySessionIdTest() {
+        //prepare
+        String id = "";
+        //when
+        byte[] captcha = captchaService.getCaptcha(id);
+        //then
+        Assertions.assertNotNull(captcha);
+    }
```
Comment (Greengrer) [inline src/test/java/com/vinylteam/vinyl/service/impl/DefaultCaptchaServiceTest.java:31]: Empty row here.

File: src/test/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServletTest.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+        when(mockedRequest.getSession()).thenReturn(mockedHttpSession);
+        when(mockedHttpSession.getId()).thenReturn("1234567890");
+        when(mockedResponse.getOutputStream()).thenReturn(mockedOutputStream);
+        //when
+        imageCaptchaServlet.doGet(mockedRequest, mockedResponse);
+        //then
+        inOrderResponse.verify(mockedResponse).setContentType("image/jpeg");
+        inOrderResponse.verify(mockedResponse).setStatus(HttpServletResponse.SC_OK);
+        inOrderRequest.verify(mockedRequest).getSession();
+    }
```
Comment (Greengrer) [inline src/test/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServletTest.java:47]: Empty row here.

File: src/test/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServletTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.mockito.InOrder;
+
+import java.io.IOException;
+
+import static org.mockito.Mockito.*;
+
+public class ImageCaptchaServletTest {
+    private final ImageCaptchaServlet imageCaptchaServlet = new ImageCaptchaServlet();
```
Comment (Greengrer) [inline src/test/java/com/vinylteam/vinyl/web/servlets/ImageCaptchaServletTest.java:18]: Empty row here.

**Review by Greengrer — COMMENTED** [review]
There is unchanged href=# and a coule of empty rows missing.

File: src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+        if (offerDocument.isPresent()) {
+            rawOffer.setShopId(5);
+            rawOffer.setRelease(getReleaseFromDocument(offerDocument.get()));
+            rawOffer.setArtist(getArtistFromDocument(offerDocument.get()));
+            rawOffer.setPrice(getPriceFromDocument(offerDocument.get()));
+            rawOffer.setCurrency(getOptionalCurrencyFromDocument(offerDocument.get()));
+            rawOffer.setGenre(getGenreFromDocument(offerDocument.get()));
+            rawOffer.setOfferLink(offerLink);
+            rawOffer.setImageLink(getHighResImageLinkFromDocument(offerDocument.get()));
+
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java]: empty row, remove plz

**Review by AlexKievDeveloper — COMMENTED** [review]
Added all changes after review

File: src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+        HashSet<RawOffer> rawOfferSet = readRawOffersFromAllOfferLinks(offerLinks);
+        return new ArrayList<>(rawOfferSet);
+    }
+
+    @Override
+    public RawOffer getRawOfferFromOfferLink(String offerLink) {
+        RawOffer rawOffer = new RawOffer();
+        Optional<Document> offerDocument = getDocument(offerLink);
+        if (offerDocument.isPresent()) {
+            rawOffer.setShopId(5);
```
Comment (littlepony99) [inline src/main/java/com/vinylteam/vinyl/util/impl/DecksParser.java]: move plz magic number 5 to local constant

**Review by Greengrer — DISMISSED** [review]
Thank you for taking my comments into account.

**Review by Greengrer — APPROVED** [review]

**Review by littlepony99 — APPROVED** [review]

**Pull request merged**